integration-tests/src/test/java/tachyon/worker/block/meta/TieredStoreIntegrationTest.java

```diff
@@ -175,31 +175,34 @@ public void unpinFileTest() throws Exception {
     Assert.assertTrue(mTFS.getInfo(file2).getInMemoryPercentage() == 100);
   }
 
-  // TODO: Add this test back when CACHE_PROMOTE is enabled again
-  /*
-   * @Test public void promoteBlock() throws IOException, InterruptedException { int fileId1 =
-   * TachyonFSTestUtils.createByteFile(mTFS, "/root/test1", WriteType.TRY_CACHE, MEM_CAPACITY_BYTES
-   * / 6); int fileId2 = TachyonFSTestUtils.createByteFile(mTFS, "/root/test2", WriteType.TRY_CACHE,
-   * MEM_CAPACITY_BYTES / 2); int fileId3 = TachyonFSTestUtils.createByteFile(mTFS, "/root/test3",
-   * WriteType.TRY_CACHE, MEM_CAPACITY_BYTES / 2);
-   *
-   * CommonUtils.sleepMs(null, CommonUtils.getToMasterHeartBeatIntervalMs(mWorkerConf) * 2 + 10);
-   *
-   * TachyonFile file1 = mTFS.getFile(fileId1); TachyonFile file2 = mTFS.getFile(fileId2);
-   * TachyonFile file3 = mTFS.getFile(fileId3);
-   *
-   * Assert.assertEquals(false, file1.isInMemory()); Assert.assertEquals(true, file2.isInMemory());
-   * Assert.assertEquals(true, file3.isInMemory()); Assert.assertEquals(MEM_CAPACITY_BYTES / 6 +
-   * MEM_CAPACITY_BYTES, mLocalTachyonCluster.getMasterInfo().getUsedBytes());
-   *
-   * InStream is = file1.getInStream(ReadType.CACHE_PROMOTE); byte[] buf = new
-   * byte[MEM_CAPACITY_BYTES / 6]; int len = is.read(buf); is.close();
-   *
-   * CommonUtils.sleepMs(null, CommonUtils.getToMasterHeartBeatIntervalMs(mWorkerConf) * 2 + 10);
-   *
-   * Assert.assertEquals(MEM_CAPACITY_BYTES / 6, len); Assert.assertEquals(true,
-   * file1.isInMemory()); Assert.assertEquals(false, file2.isInMemory()); Assert.assertEquals(true,
-   * file3.isInMemory()); Assert.assertEquals(MEM_CAPACITY_BYTES / 6 + MEM_CAPACITY_BYTES,
-   * mLocalTachyonCluster.getMasterInfo().getUsedBytes()); }
-   */
+  @Test
+  public void promoteBlock() throws IOException, InterruptedException {
+    TachyonFile file1 =
+        TachyonFSTestUtils.createByteFile(mTFS, "/root/test1", TachyonStorageType.STORE,
+            UnderStorageType.PERSIST, MEM_CAPACITY_BYTES / 6);
+    TachyonFile file2 =
+        TachyonFSTestUtils.createByteFile(mTFS, "/root/test2", TachyonStorageType.STORE,
+            UnderStorageType.PERSIST, MEM_CAPACITY_BYTES / 2);
+    TachyonFile file3 =
+        TachyonFSTestUtils.createByteFile(mTFS, "/root/test3", TachyonStorageType.STORE,
+            UnderStorageType.PERSIST, MEM_CAPACITY_BYTES / 2);
+
+    CommonUtils.sleepMs(null, mWorkerToMasterHeartbeatIntervalMs * 3);
+
+    Assert.assertEquals(0, mTFS.getInfo(file1).getInMemoryPercentage());
```

Comment:
Maybe we can iterate through these three files to see whose InMemoryPercentage is not 100 and then read it via promote?